libursa/src/cl/verifier.rs

```diff
@@ -45,16 +46,26 @@ impl Verifier {
     pub fn new_proof_verifier() -> UrsaCryptoResult<ProofVerifier> {
         Ok(ProofVerifier {
             credentials: Vec::new(),
+            common_attributes: HashMap::new(),
         })
     }
 }
 
 #[derive(Debug)]
 pub struct ProofVerifier {
     credentials: Vec<VerifiableCredential>,
+    common_attributes: HashMap<String, Option<BigNumber>>,
```

Comment:
Why is this anything but a Vec<string>?

Reply:
The verifier's objective is to ensure that common attributes have same `m_hat`s. So when the verifier processes the first sub-proof, it stores the `m_hat` for corresponding common attributes in the hashmap. For the subsequent attributes, it checks that common attributes have same value.

Reply:
Okay so why is it Option<BigNumber> and not BigNumber?

Reply:
Because the verifier specifies the attribute names before it starts verifying the proof. So their value is `None` at that time. A simpler but less efficient implementation would be to use a `HashSet<String>` of attribute names which is filled during calls to `add_common_attribute` and a `HashMap<String, BigNumber>` which is filled during call to `verify`